### PR TITLE
Moved database instance rule classes into a test support module

### PR DESCRIPTION
**Pull request by bradbeck**
related:

https://github.com/sonatype/nexus-pro/pull/557

### CODE REVIEW

**Comment by jdillon** [comment]
junit and jul deps in nexus-orient can be removed if we do this.

ftr i'm not sure its worth an extra module just for this, but /me shrugs

File: testsupport/nexus-test-db/src/main/java/org/sonatype/nexus/test/orient/DatabaseInstanceRule.java [hunk 1]
```diff
@@ -10,17 +10,21 @@
...
 
 import com.google.common.base.Throwables;
 import com.google.common.collect.Lists;
+
 import org.junit.rules.ExternalResource;
 import org.junit.runners.model.MultipleFailureException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.slf4j.bridge.SLF4JBridgeHandler;
+import org.sonatype.nexus.orient.DatabaseInstance;
```
Comment (mcculls) [inline testsupport/nexus-test-db/src/main/java/org/sonatype/nexus/test/orient/DatabaseInstanceRule.java]: import order is off here...

File: components/nexus-core/src/main/java/org/sonatype/nexus/internal/orient/LogConfigurationCustomizerImpl.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
 {
   @Override
   public void customize(final Configuration config) {
-    config.setLoggerLevel("org.sonatype.nexus.orient", DEFAULT);
+    config.setLoggerLevel("org.sonatype.nexus.test.orient", DEFAULT);
```
Comment (mcculls) [inline components/nexus-core/src/main/java/org/sonatype/nexus/internal/orient/LogConfigurationCustomizerImpl.java]: side-effect of the refactoring... this should be left as "org.sonatype.nexus.orient" as this is non-test code

File: testsupport/nexus-test-db/pom.xml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  <dependencies>
+    <dependency>
+      <groupId>org.sonatype.nexus</groupId>
+      <artifactId>nexus-orient</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <optional>true</optional>
```
Comment (mcculls) [inline testsupport/nexus-test-db/pom.xml]: these were optional in nexus-orient to avoid them leaking onto the non-test classpath of other modules, but they can be made non-optional here.

**Comment by bradbeck** [comment]
http://bamboo.s/browse/NX3-OSSF377-4

File: components/nexus-orient/pom.xml [hunk 4]
```diff
@@ -62,15 +62,9 @@
...
-      <artifactId>junit</artifactId>
-      <optional>true</optional>
-    </dependency>
-
-    <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>jul-to-slf4j</artifactId>
-      <optional>true</optional>
+      <scope>test</scope>
     </dependency>
```
Comment (jdillon) [inline components/nexus-orient/pom.xml]: is jul-to-slf4j needed anymore in nexus-orient now?

**Comment by jdillon** [comment]
+1 with minor comment about dependency tidy

File: components/nexus-orient/pom.xml (hunk 4 above)
Comment (mcculls) [inline components/nexus-orient/pom.xml]: There are some "Trial" test classes which refer to `SLF4JBridgeHandler`, but I think that might be down to copy+paste. Certainly they aren't run as part of the tests so I think the references to `SLF4JBridgeHandler.install` can be removed without disturbing anything else.

Comment (bradbeck) [inline components/nexus-orient/pom.xml]: Several of the test classes (e.g. `BackupRestoreTrial`) rely on `org.slf4j.bridge.SLF4JBridgeHandler` which appears to be supplied by `jul-to-slf4j`.

Comment (jdillon) [inline components/nexus-orient/pom.xml]: ic, minor downside to this change, is that tests in nexus-orient can't use this rule.  granted tests are a bit light atm, but this does mean if we add more tests we may have to duplicate those rules, or find other ways to do the same thing.  BackupRestoreTrial for example, probably could have been changed to use the persistent rule.

the rules came much later than those, and were never updated.  I wonder if they were updated, if we would still think extracting the rules was a good idea?
